### PR TITLE
(PE-36269) ensure proper CA issuer is set in renewed certificates

### PR DESCRIPTION
**Pull request by jonathannewman**
This updates the issuer for the generated CRL to ensure it has the same value as the original CRL.

### CODE REVIEW

File: test/unit/puppetlabs/puppetserver/certificate_authority_test.clj [hunk 1]
```diff
@@ -2233,6 +2234,10 @@
     (ca/overwrite-existing-crl! crl rest-of-full-chain (:capub settings) (:cakey settings) ca-cert (vec all-serials) (:cacrl settings))
     (let [[updated-crl & _rest-of-full-chain] (utils/pem->crls (:cacrl settings))
           updated-crl-number (utils/get-crl-number updated-crl)]
+      ;; currently the CA certs only have one cert, and not an intermediate
+      ;; so while this test is a valid test, it is very unlikely to fail.
```
Comment (justinstoller) [inline test/unit/puppetlabs/puppetserver/certificate_authority_test.clj:2238]: It looks like there's a "bundle-cadir" that has a small CA chain in it that we can use. See usage here: https://github.com/puppetlabs/puppetserver/pull/2791/files#diff-5dee42180e003e4d76fbca7cddd079fe43ce975631592037eab3d19bf04af052R489

**Comment by jonathannewman** [comment]
Updated to use the `bundle-ca`.  I had to add a ca_pub file which was copied over from the `master` directory in the test space.

**Comment by jonathannewman** [comment]
Test changes introduced fail prior to the fix being put in place.

File: src/clj/puppetlabs/puppetserver/certificate_authority.clj [hunk 2]
```diff
@@ -2363,7 +2363,8 @@
         ;; the "next update" and "crl-number" will get updated by the process that
         ;; adds the serial numbers to the crl
         new-full-crl (utils/generate-crl
-                       (.getIssuerX500Principal cacert)
+                       ;; ensure the issuer is identical
+                       (.getIssuerX500Principal crl)
```
Comment (jonathannewman) [inline src/clj/puppetlabs/puppetserver/certificate_authority.clj:2367]: The "issuer" for the `cacert` ends up being the root cert.  In this case, we want to use the same issuer as the original CRL, so it matches.

File: test/unit/puppetlabs/puppetserver/certificate_authority_test.clj [hunk 3]
```diff
@@ -2133,7 +2134,7 @@
               public-key (utils/pem->public-key (:capub settings))
               private-key (utils/pem->private-key (:cakey settings))
               new-crl (utils/generate-crl
-                        (.getIssuerX500Principal (utils/pem->ca-cert (:cacert settings) (:cakey settings)))
+                        (.getSubjectX500Principal (utils/pem->ca-cert (:cacert settings) (:cakey settings)))
```
Comment (jonathannewman) [inline test/unit/puppetlabs/puppetserver/certificate_authority_test.clj:2137]: This is the alternative to getting the issuer from the crl.

File: test/unit/puppetlabs/puppetserver/certificate_authority_test.clj [hunk 4]
```diff
@@ -2154,7 +2155,7 @@
           (is (ca/crl-expires-in-n-days? path-as-string settings 6))))))
 
 (deftest overwrite-existing-crl!-test
-  (let [settings (testutils/ca-sandbox! cadir)
+  (let [settings (testutils/ca-sandbox! bundle-cadir)
```
Comment (jonathannewman) [inline test/unit/puppetlabs/puppetserver/certificate_authority_test.clj:2158]: the `bundle-cadir` has a root CA and an intermediate, so changing this allows that path to be tested.

File: test/unit/puppetlabs/puppetserver/certificate_authority_test.clj [hunk 5]
```diff
@@ -2233,6 +2234,8 @@
     (ca/overwrite-existing-crl! crl rest-of-full-chain (:capub settings) (:cakey settings) ca-cert (vec all-serials) (:cacrl settings))
     (let [[updated-crl & _rest-of-full-chain] (utils/pem->crls (:cacrl settings))
           updated-crl-number (utils/get-crl-number updated-crl)]
+      (is (= (.getName (.getIssuerX500Principal ^X509CRL updated-crl))
```
Comment (jonathannewman) [inline test/unit/puppetlabs/puppetserver/certificate_authority_test.clj:2237]: This tests the defect.  Without the change to use the issuer from the crl in `certificate_authority` this test fails.

File: src/clj/puppetlabs/puppetserver/certificate_authority.clj [hunk 6]
```diff
@@ -824,7 +824,7 @@
 (schema/defn expired-inventory-serials :- [BigInteger]
   [{:keys [cert-inventory inventory-lock inventory-lock-timeout-seconds]} :- CaSettings]
   (common/with-safe-read-lock inventory-lock inventory-lock-descriptor inventory-lock-timeout-seconds
-    (log/trace (i18n/trs "Extracting expired serials from inventory file {1}" cert-inventory))
+    (log/trace (i18n/trs "Extracting expired serials from inventory file {0}" cert-inventory))
```
Comment (jonathannewman) [inline src/clj/puppetlabs/puppetserver/certificate_authority.clj:827]: This was a typo.

**Review by rileynewton — APPROVED** [review]
👍

**Pull request merged**